### PR TITLE
improve clarity of certificate usage guides

### PR DESCRIPTION
**Pull request by derisen**
* improve clarity in certificate usage docs
* update certificate usage samples
* update example certificate files

cc @yuazhou

### CODE REVIEW

File: lib/msal-node/docs/certificate-credentials.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 You can build confidential client applications with MSAL Node (web apps, daemon apps etc). A **client credential** is mandatory for confidential clients. Client credential can be a:
 
-* `clientSecret`: a secret string generated set on the app registration.
+* `clientSecret`: a secret string generated on the app registration.
```
Comment (peterzenz) [inline lib/msal-node/docs/certificate-credentials.md]: "generated on the app registration"  consider:   a secret string generated during the app registration, or updated post registration for an existing application.

File: lib/msal-node/docs/certificate-credentials.md [hunk 2]
```diff
@@ -21,10 +21,11 @@ Afterwards, add the path to `OpenSSL` to your **environment variables** so that
 Type the following in a terminal. You will be prompted to enter a *pass phrase* (this is optional, but recommended):
 
 ```bash
-    openssl req -x509 -newkey rsa:2048 -sha256 -days 365 -keyout example.key -out example.crt -subj "/CN=example.com"
+    openssl req -x509 -newkey rsa:2048 -keyout example.key -out example.crt -subj "/CN=example.com"
 ```
 
-> :lightbulb: add *-nodes* if you don't want to encrypt your private key with a *pass phrase*.
+> :bulb: Add the *-nodes* parameter above if you don't want to encrypt your private key with a *pass phrase*.
+> :bulb: By default, the self signed certificate will be valid for 32 days. Use the *-days* parameter to specify another number.
```
Comment (peterzenz) [inline lib/msal-node/docs/certificate-credentials.md]: Consider:   "The default lifetime of a self-signed certificate is 32 days.  The *-days* parameter can be used to change this."

File: lib/msal-node/docs/certificate-credentials.md [hunk 3]
```diff
@@ -35,34 +36,24 @@ In your terminal, you should see:
...
-    ----- 
+    -----
 ```
 
 After that, the following files should be generated:
 
-* `example.crt`: your public key. This is the actual certificate file that you'll mainly work with.
-* `example.key`: your private key
+* `example.crt`: your public key. This is the actual certificate file that you'll upload to Azure AD.
+* `example.key`: your private key.
```
Comment (peterzenz) [inline lib/msal-node/docs/certificate-credentials.md]: Do we have a link we can point to on private key best practices we can link to here?

File: lib/msal-node/docs/certificate-credentials.md [hunk 4]
```diff
@@ -104,25 +95,13 @@ z2HCpDsa7dxOsKIrm7F1AtGBjyB0yVDjlh/FA7jT5sd2ypBh3FVsZGJudQsLRKfE
...
-```
-
-You'll be prompted for the pass phrase you specified earlier and a new encryption password:
-
-```console
-Enter pass phrase for example.key.pem:
-Enter Encryption Password:
-Verifying - Enter Encryption Password:
-```
+> :information_source: Alternatively, your private key may begin with `-----BEGIN PRIVATE KEY-----` (unencrypted *PKCS#8*) or `-----BEGIN RSA PRIVATE KEY-----` (*PKCS#1*). These formats are also permissible. If you need, you can ensure that your private key is in *PKCS#8* format by typing:
```
Comment (peterzenz) [inline lib/msal-node/docs/certificate-credentials.md]: Consider:  "The following can be used to convert any compatible key to the *PKCS#8* key type.

File: lib/msal-node/docs/certificate-credentials.md [hunk 5]
```diff
@@ -191,36 +141,48 @@ const forge = require('node-forge');
...
 
+### (Optional) Creating an HTTPS server
+
+The OAuth 2.0 protocol recommends using an HTTPS connection whenever possible. Most cloud services like Azure App Service will provide HTTPS connection by default via proxy. If for testing purposes you would like to setup your own HTTPS server, see the [Node.js HTTPS guide](https://nodejs.org/knowledge/HTTP/servers/how-to-create-a-HTTPS-server/).
+
+You'll also need to add your self-signed certificates to the *credential manager* / *key chain* of your **OS** to bypass the browser's security policy. You may still see a warning in your browser afterwards (e.g. Chrome).
+
+* For Windows users, follow the guide here: [How to: View certificates with the MMC snap-in](https://docs.microsoft.com/dotnet/framework/wcf/feature-details/how-to-view-certificates-with-the-mmc-snap-in).
+
+* For Linux and MacOS users, use community guides -with judgment- to learn about how to install certificates.
```
Comment (peterzenz) [inline lib/msal-node/docs/certificate-credentials.md]: Consider: "For Linux and MacOS users, please consult your operating system documentation on how to install certificates."

File: lib/msal-node/docs/key-vault-managed-identity.md [hunk 6]
```diff
@@ -17,18 +17,10 @@ Now you can import your certificate to Key Vault. **Azure Key Vault** expects ce
 * *.pem* file format contains one or more X509 certificate files.
 * *.pfx* file format is an archive file format for storing several cryptographic objects in a single file i.e. server certificate (issued for your domain), a matching private key, and may optionally include an intermediate CA.
 
-> :bulb: If you don't have any certificates at hand, you can use Azure Key Vault to generate it for you. It will have the additional benefits of assigning partner Certificate Authority and automating certificate rotation. For more information, see [Quickstart: Generate a certificate with Azure Key Vault using the Azure portal](https://docs.microsoft.com/azure/key-vault/certificates/quick-create-portal)
+> :bulb: If you don't have any certificates at hand, you can use Azure Key Vault to generate it for you. It will have the additional benefits of assigning a partner Certificate Authority and automating certificate rotation. For more information, see [Quickstart: Generate a certificate with Azure Key Vault using the Azure portal](https://docs.microsoft.com/azure/key-vault/certificates/quick-create-portal)
```
Comment (peterzenz) [inline lib/msal-node/docs/key-vault-managed-identity.md]: Consider: "you can use Azure Key Vault to generate one for you."

**Review by peterzenz — APPROVED** [review]
Minor wordsmithing comments, up to you if you want to take them.  Thanks!

File: lib/msal-node/docs/certificate-credentials.md (hunk 2 above)
Comment (derisen) [reply lib/msal-node/docs/certificate-credentials.md]: done

File: lib/msal-node/docs/certificate-credentials.md (hunk 3 above)
Comment (derisen) [reply lib/msal-node/docs/certificate-credentials.md]: Nothing that I could find, unfortunately.

File: lib/msal-node/docs/certificate-credentials.md (hunk 4 above)
Comment (derisen) [reply lib/msal-node/docs/certificate-credentials.md]: done

File: lib/msal-node/docs/certificate-credentials.md (hunk 5 above)
Comment (derisen) [reply lib/msal-node/docs/certificate-credentials.md]: done

File: lib/msal-node/docs/key-vault-managed-identity.md (hunk 6 above)
Comment (derisen) [reply lib/msal-node/docs/key-vault-managed-identity.md]: done

File: lib/msal-node/docs/certificate-credentials.md (hunk 1 above)
Comment (derisen) [reply lib/msal-node/docs/certificate-credentials.md]: done

**Review by tnorling — APPROVED** [review]

**Pull request merged**